File: 2021/py/day1.py
import helpers

measurements = helpers.list_int_from_file('2021/inputs/day1')

# part one
print(len([i for i in range(len(measurements)-1) if measurements[i] < measurements[i+1]]))

# part two
print(len([i for i in range(len(measurements)-3) if sum(measurements[i:i+3]) < sum(measurements[i+1:i+4])]))

File: 2015/cs/Day1.cs
namespace Runner
{
    internal static class Day1
    {
        internal static void Run()
        {
            // part one
            var line = File.ReadAllLines("../inputs/day1")[0];
            Console.WriteLine(line.Count(c => c == '(') - line.Count(c => c == ')'));
            
            // part two
            var floor = 0;
            var index = 0;
            foreach (char c in line)
            {
                index += 1;
                floor += c == '(' ? 1 : -1;
                if (floor == -1) break; 
            }

            System.Console.WriteLine(index);
        }
    }
}